Description: Modify doc/make.py to add sphinxopts options.
 Allow pass the value of sphinxopts command line option to sphinx-build.
Author: Juan Picca <jumapico@gmail.com>
Last-Update: 2015-06-27
---
--- a/doc/make.py
+++ b/doc/make.py
@@ -5,13 +5,13 @@ import os
 import shutil
 import sys
 
-def html():
-    os.system('sphinx-build -b html -d build/doctrees . build/html')
+def html(opts):
+    os.system('sphinx-build %s -b html -d build/doctrees . build/html' % opts.get('sphinxopts', ''))
 
-def latex():
+def latex(opts):
     if sys.platform != 'win32':
         # LaTeX format.
-        os.system('sphinx-build -b latex -d build/doctrees . build/latex')
+        os.system('sphinx-build %s -b latex -d build/doctrees . build/latex' % opts.get('sphinxopts', ''))
 
         # Produce pdf.
         os.chdir('build/latex')
@@ -27,12 +27,12 @@ def latex():
     else:
         print 'latex build has not been tested on windows'
 
-def clean():
+def clean(opts):
     shutil.rmtree('build')
 
-def all():
-    html()
-    latex()
+def all(opts):
+    html(opts)
+    latex(opts)
 
 
 funcd = {
@@ -44,11 +44,16 @@ funcd = {
 
 
 if len(sys.argv)>1:
+    opts = {}
     for arg in sys.argv[1:]:
+        if arg.startswith('--sphinxopts='):
+            opts['sphinxopts'] = arg.replace('--sphinxopts=', '')
+            continue
+
         func = funcd.get(arg)
         if func is None:
             raise SystemExit('Do not know how to handle %s; valid args are'%(
                     arg, funcd.keys()))
-        func()
+        func(opts)
 else:
-    all()
+    all({})
